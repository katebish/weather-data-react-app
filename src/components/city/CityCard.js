import { capitalize } from 'lodash';
import DeleteIcon from '@mui/icons-material/Delete';
import RefreshIcon from '@mui/icons-material/Refresh';
import Card from '@mui/material/Card';
import CardActions from '@mui/material/CardActions';
import CardContent from '@mui/material/CardContent';
import IconButton from '@mui/material/IconButton';
import Typography from '@mui/material/Typography';
import { PropTypes } from 'prop-types';

const CityCard = props => {
    const {
        name,
        description,
        temp,
        feelsLike,
        reloadError,
        deleteCity,
        reloadWeatherData
    } = props;

  return (
    <Card sx={{ minWidth: 275 }}>
      <CardContent>
        <Typography variant='h5'>
        {name}
        </Typography>
        <Typography color="text.secondary">
        {capitalize(description)}
        </Typography>
        <Typography color="text.secondary" variant="body2">
        {`Temp: ${temp}`}
        </Typography>
        <Typography color="text.secondary" variant="body2">
        {`Feels like: ${feelsLike}`}
        </Typography>
      </CardContent>
      <CardActions>
        <IconButton aria-label="delete" onClick={deleteCity}>
            <DeleteIcon size="small"/>
        </IconButton>
        <IconButton aria-label="refresh" onClick={reloadWeatherData}>
            <RefreshIcon size="small"/>
        </IconButton>
        {reloadError && <Typography color="red">Failed to reload</Typography>}
      </CardActions>
    </Card>
  );
};

CityCard.defaultProps = {
    reloadError: false
};

CityCard.propTypes = {
    name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    temp: PropTypes.number.isRequired,
    feelsLike: PropTypes.number.isRequired,
    reloadError: PropTypes.bool,
    deleteCity: PropTypes.func.isRequired,
    reloadWeatherData: PropTypes.func.isRequired
};

export default CityCard;

