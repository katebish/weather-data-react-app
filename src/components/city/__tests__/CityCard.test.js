import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import CityCard from '../CityCard';

const mockProps = {
    name: 'London',
    description: 'cloudy',
    temp: 12,
    feelsLike: 13,
    deleteCity: jest.fn(),
    reloadWeatherData: jest.fn()
}

const renderCityCard = additionalProps => {
    const props = { ...mockProps, ...additionalProps };
    return render(<CityCard {...props} />);
};

const queryReloadErrorMessage = () => screen.queryByText('Failed to reload');

describe('<CityCard />', () => {
    it('matches snapshot', () => {
        const { baseElement } = renderCityCard();
        expect(baseElement).toMatchSnapshot();
    });
    it('calls deleteCity on clicking delete button', () => {
        renderCityCard();
        userEvent.click(screen.getByTestId('delete_button'));
        expect(mockProps.deleteCity).toHaveBeenCalled();
    });
    it('calls reloadWeatherData on clicking delete button', () => {
        renderCityCard();
        userEvent.click(screen.getByTestId('reload_button'));
        expect(mockProps.reloadWeatherData).toHaveBeenCalled();
    });
    it('does not render "Failed to reload" error message if reloadError is false', () => {
        renderCityCard({ reloadError: false });
        expect(queryReloadErrorMessage()).not.toBeInTheDocument();
    });
    it('renders "Failed to reload" error message with red text color if reloadError is true', () => {
        renderCityCard({ reloadError: true });
        const reloadErrorMessage = queryReloadErrorMessage();
        expect(reloadErrorMessage).toBeInTheDocument();
        expect(reloadErrorMessage).toHaveStyle({ color: 'red' });
    });
});
