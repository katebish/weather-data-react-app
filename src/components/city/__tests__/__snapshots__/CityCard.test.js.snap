// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CityCard /> matches snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-pwngrv-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          London
        </h5>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
        >
          Cloudy
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          Temp: 12
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          Feels like: 13
        </p>
      </div>
      <div
        class="MuiCardActions-root MuiCardActions-spacing css-dnrpxu-MuiCardActions-root"
      >
        <button
          aria-label="delete"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-testid="delete_button"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="DeleteIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="reload"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-testid="reload_button"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="RefreshIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;
